src/box/iproto.cc

```diff
@@ -1684,15 +1732,54 @@ tx_accept_wpos(struct iproto_connection *con, const struct iproto_wpos *wpos)
 	}
 }
 
+/**
+ * Since the processing of requests within a transaction
+ * for a stream can occur in different fibers, we store
+ * a pointer to transaction in the stream structure.
+ * Check if message belongs to stream and there is active
+ * transaction for this stream. In case it is so, sets this
+ * transaction for current fiber.
+ */
+static inline struct txn *
+tx_prepare_transaction_for_request(struct iproto_msg *msg)
+{
+	struct txn *txn = NULL;
+	if (msg != NULL && msg->stream != NULL) {
+		txn = msg->stream->txn;
+		/*
+		 * When we do any operations (which are written to `wal`)
+		 * outside of transaction, we consider each such operation
+		 * as a small transaction and write it to `wal` immediately.
+		 * When operation is performed as part of transaction, we
+		 * write all transaction to `wal` at the commit. In this case,
+		 * `is_commit` flag will be set when writing to `wal` for the
+		 * last operation in transaction, the rest operations must have
+		 * this flag set to false, to mark that they all belongs to the
+		 * same transaction.
+		 */
+		if (txn != NULL)
+			msg->header.is_commit = false;
```

Comment:
This check and flag set can be done in the iproto threads.

Reply:
Yes move it to iproto_msg_set_stream

Reply:
After private discussion we agreed to leave it as it is